content/en/docs/Concepts/olm-architecture/dependency-resolution/_index.md

```diff
@@ -80,6 +111,169 @@ The resolver sees these constraints as (assuming this is for `FooOperator v1.0.0
 
 This looks (and is) straightforward, but it's worth looking at how this is handled at runtime in [Understanding Preferences](#understanding-preferences).
 
+### Generic Constraint
+
+An `olm.constraint` property declares a dependency constraint of a particular type, differentiating non-constraint and constraint properties. Its `value` field is an object containing a `failureMessage` field holding a string-representation of the constraint message that will be surfaced to the users if the constraint is not satisfiable at runtime, and as an informative comment for readers, and exactly one of the following keys that denotes the constraint type:
+
+- `gvk`: type whose `value` and interpretation is identical to `olm.gvk`.
+- `package`: type whose `value` and interpretation is identical to `olm.package`.
+- `cel`: a [Common Expression Language (CEL)](https://github.com/google/cel-go) expression evaluated at runtime by OLM's resolver over arbitrary bundle properties and cluster information ([enhancement][cel-ep]).
+- `all`, `any`, `none`: conjunction, disjunction, and negation constraints, respectively, containing one or more concrete constraints, ex. `gvk`, or a nested compound constraint ([enhancement][compound-ep]).
+
+#### Common Expression Language (CEL)
+
+The `cel` struct is specific to CEL constraint type that supports CEL as the expression language. The `cel` struct has `rule` field which contains the CEL expression string that will be evaluated against operator properties at the runtime to determine if the operator satisfies the constraint.
+
+For example:
+```yaml
+type: olm.constraint
+value:
+    failureMessage: 'require to have "certified"'
+    cel:
+        rule: 'properties.exists(p, p.type == "certified")'
```

Comment:
Quick note/nit: when viewing the rendered deploy preview, the indentation of this example doesn't match the rest of the examples that have been documented. This also applies to the other CEL example documented in this header.